### PR TITLE
Revert "Remove DependencyModel dependency"

### PR DESCRIPTION
**Pull request by pranavkm**
This reverts commit 4299a4d50bd782b9253c0e69a0505463492bfc5f.

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @pakrym \ @davidfowl

File: src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/DefaultAssemblyProvider.cs [hunk 1]
```diff
@@ -75,9 +76,12 @@ public IEnumerable<Assembly> CandidateAssemblies
                     return new[] { _entryAssembly };
                 }
 
+                var candidateLibraries = GetCandidateLibraries();
```
Comment (pakrym) [inline src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/DefaultAssemblyProvider.cs]: Do you use this? And next one?

**Comment by pakrym** [comment]
Small comment then :shipit:

File: src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/DefaultAssemblyProvider.cs (hunk 1 above)
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/DefaultAssemblyProvider.cs]: Nope, I'll remove these.

**Pull request closed without merging**